app/models/conversion_host.rb

```diff
@@ -91,7 +95,7 @@ def verify_credentials(auth_type = nil, options = {})
   #  - The number of concurrent tasks has not reached the limit.
   #
   def eligible?
-    source_transport_method.present? && verify_credentials && check_concurrent_tasks
+    source_transport_method.present? && authentication_check('v2v').first && check_concurrent_tasks
```

Comment:
this throws me a little.

In `verify_credentials`, the connect code uses v2v authentications || the first one. This logic looks different.

I liked having some high level function in here like `verify_credentials`. This middle clause has me wondering if `authentication_check.first` is right. Or what about the check when there are no authentications that seems to happen. So those low level question seems out of place in this high level method.

Reply:
Internally `authentication_check` calls `verify_credentials`, it's essentially an interface method that classes should define on their own.

Reply:
I read the method, and understand why this is true.
But reading the old code for some reason felt higher level

could we add a readability method?

```ruby
def authentication_valid_for?(*args)
  authentication_check(*args).first
end
```

Reply:
I get what you're saying but I'd like to keep it minimalist for now, since that would have to be added to the Authentication mixin.

Reply:
I think @agrare's suggestion to use `authentication_status_ok?` accomplishes what you were after, and I've updated the PR to use it.